hyperspace/cosmos/src/key_provider.rs

```diff
@@ -0,0 +1,74 @@
+use super::CosmosClient;
+use crate::error::Error;
+use bech32::{ToBase32, Variant};
+use bitcoin::{
+	hashes::hex::ToHex,
+	util::bip32::{ExtendedPrivKey, ExtendedPubKey},
+};
+use ibc::core::ics24_host::identifier::ChainId;
+use primitives::KeyProvider;
+use serde::{Deserialize, Serialize};
+use std::{path::Path, str::FromStr};
+use tendermint::account::Id as AccountId;
+
+pub const KEYSTORE_DEFAULT_FOLDER: &str = ".hermes/keys/";  //TODO: Should be adjusted to the correct path later
+pub const KEYSTORE_DISK_BACKEND: &str = "keyring-test";
+pub const KEYSTORE_FILE_EXTENSION: &str = "json";
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct KeyEntry {
+	/// Public key
+	pub public_key: ExtendedPubKey,
+
+	/// Private key
+	pub private_key: ExtendedPrivKey,
+
+	/// Account Bech32 format - TODO allow hrp
+	pub account: String,
+
+	/// Address
+	pub address: Vec<u8>,
```

Comment:
Does it make sense having an `Address` type?

Reply:
Let's keep it for now. It may be used in other unimplemented methods, Otherwise I'll drop it.